### PR TITLE
ENH: add dtype.ndim

### PR DESCRIPTION
**Pull request by eric-wieser**
Also: Convert `len\((.*?)\.shape\)` to `\1.ndim`

### CODE REVIEW

File: numpy/add_newdocs.py [hunk 1]
```diff
@@ -6355,6 +6355,13 @@ def luf(lamdaexpr, *args, **kwargs):
 
     """))
 
+add_newdoc('numpy.core.multiarray', 'dtype', ('ndim',
+    """
+    Number of dimensions of the sub-array if this data type describes a
+    sub-array, and ``0`` otherwise.
+
```
Comment (juliantaylor) [inline numpy/add_newdocs.py]: needs `.. versionadded:: 1.13.0`

**Review by juliantaylor — COMMENTED** [review]
hm probably makes sense, should be mentioned in release notes.

**Comment by eric-wieser** [comment]
@juliantaylor: Ok, done both those things, and it's green again

**Comment by juliantaylor** [comment]
thanks

**Pull request merged**